platform/platform.go

```diff
@@ -407,18 +426,75 @@ func (p *Platform) configureQuota(state DeploymentState) error {
 	return nil
 }
 
+func (p *Platform) configureHealthCheck(state *DeploymentState) error {
+	if needToConfigureHealthCheck(state) {
+		step := (*state.sg).Add("Configuring health check...")
+
+		processes, err := state.client.GetApplicationProcesses(state.app.GUID)
+		if err != nil {
+			step.Abort()
+			return fmt.Errorf("failed to get application processes: %v", err)
+		}
+
+		if len(processes) == 0 {
+			step.Abort()
+			return fmt.Errorf("no processes found")
+		}
+
+		for _, process := range processes {
+			process.HealthCheckType = state.healthCheckParams.Type
+
+			if state.healthCheckParams.Endpoint != "" {
+				process.HealthCheckEndpoint = state.healthCheckParams.Endpoint
+			}
+
+			if state.healthCheckParams.InvocationTimeout != 0 {
+				process.HealthCheckInvocationTimeout = state.healthCheckParams.InvocationTimeout
+			}
+
+			if state.healthCheckParams.Timeout != 0 {
+				process.HealthCheckTimeout = state.healthCheckParams.Timeout
+			}
+
+			p.log.Debug("updating process: %v", process)
+
+			_, err = state.client.UpdateApplicationProcess(process)
+			if err != nil {
+				step.Abort()
+				return fmt.Errorf("unable to configure application process: %v", err)
+			}
+		}
+
+		step.Done()
+	}
+	return nil
+}
+
+func needToConfigureHealthCheck(state *DeploymentState) bool {
+	return state.healthCheckParams != nil && (state.healthCheckParams.Endpoint != "" ||
+		state.healthCheckParams.Type != "" || state.healthCheckParams.InvocationTimeout != 0 ||
```

Comment:
Are you sure it's needed? I would only check if `state.healthCheckParms` is not `nil`.

Reply:
I guess we can limit the check to only top level field. There's validation later, that would catch other problems anyway.

Reply:
Awesome! I'll then simplify this part and remove entirely this function :)

Thanks!

Reply:
Did it already :-)